sktime/forecasting/base/_base.py

```diff
@@ -1878,6 +1888,25 @@ def _check_fh(self, fh):
                     "horizon, please re-fit the forecaster. " + msg
                 )
             # if existing one and new match, ignore new one
+        in_sample_pred = (
+            self.get_tag("capability:insample")
+            if not pred_int
+            else self.get_tag("capability:pred_int:insample")
+        )
+        if (
+            not in_sample_pred
+            and self._fh is not None
+            and cutoff is not None
+            and not self._fh.is_all_out_of_sample(cutoff)
+        ):
+            msg = (
+                f"in-sample prediction is currently not supported: "
```

Comment:
I would put the forecaster name more early in the error message.

Reply:
done